types/_class_actor_functions.lua

```diff
@@ -0,0 +1,506 @@
+---@param actorIndex integer @The index of the actor.
+---@param spellIndex integer @The index of the spell.
+---@return boolean @Whether a player is the caster of the spell.
+function api.DoesActorSpellsActiveHavePlayerCaster(actorIndex, spellIndex) end
+
+---Get the player ID of the caster of the spell at a certain index in an actor's latest spells active changes.
+---@param actorIndex integer @The index of the actor.
+---@param spellIndex integer @The index of the spell.
+---@return integer @The player ID of the caster.
+function api.GetActorSpellsActiveCasterPid(actorIndex, spellIndex) end
+
+---Get the refId of the actor caster of the spell at a certain index in an actor's latest spells active changes.
+---@param actorIndex integer @The index of the actor.
+---@param spellIndex integer @The index of the spell.
+---@return string @The refId of the caster.
+function api.GetActorSpellsActiveCasterRefId(actorIndex, spellIndex) end
+
+---Get the refNum of the actor caster of the spell at a certain index in an actor's latest spells active changes.
+---@param actorIndex integer @The index of the actor.
+---@param spellIndex integer @The index of the spell.
+---@return integer @The refNum of the caster.
+function api.GetActorSpellsActiveCasterRefNum(actorIndex, spellIndex) end
+
+---Get the mpNum of the actor caster of the spell at a certain index in an actor's latest spells active changes.
+---@param actorIndex integer @The index of the actor.
+---@param spellIndex integer @The index of the spell.
+---@return integer @The mpNum of the caster.
+function api.GetActorSpellsActiveCasterMpNum(actorIndex, spellIndex) end
+
+---Check whether there is any positional data for the actor at a certain index in the read actor list.
+---
+---This is only useful when reading the actor list data recorded for a particular cell.
+---@param index integer @The index of the actor.
+---@return boolean @Whether the read actor list contains positional data.
+function api.DoesActorHavePosition(index) end
+
+---Check whether there is any dynamic stats data for the actor at a certain index in the read actor list.
+---
+---This is only useful when reading the actor list data recorded for a particular cell.
+---@param index integer @The index of the actor.
+---@return boolean @Whether the read actor list contains dynamic stats data.
+function api.DoesActorHaveStatsDynamic(index) end
+
+---Set the cell of the temporary actor list stored on the server.
+---
+---The cell is determined to be an exterior cell if it fits the pattern of a number followed by a comma followed by another number.
+---@param cellDescription string @The description of the cell.
+function api.SetActorListCell(cellDescription) end
+
+---Set the action type of the temporary actor list stored on the server.
+---@param action string @The action type (0 for SET, 1 for ADD, 2 for REMOVE, 3 for REQUEST).
```

Comment:
Can you add an alias for this argument? These action types are also used in other places.